public/app/plugins/datasource/prometheus/querybuilder/shared/parsingUtils.ts

```diff
@@ -0,0 +1,179 @@
+import { SyntaxNode, TreeCursor } from '@lezer/common';
+
+// This is used for error type for some reason
+export const ErrorName = '⚠';
+
+export function getLeftMostChild(cur: SyntaxNode): SyntaxNode | null {
```

Comment:
nit, but it seems like this could be simplified:
```suggestion
export function getLeftMostChild(cur: SyntaxNode): SyntaxNode {
  return cur.firstChild ? getLeftMostChild(cur.firstChild) : cur;
}
```

Reply:
Nice yeah that makes sense.